### PR TITLE
deduplicate benchmarks

### PR DESCRIPTION
**Pull request by ermalkaleci**
Move benchmarks into separate package. Benchmarks are wrapped into macros that gets expanded into different runtime benchmarks

### CODE REVIEW

File: modules/nft/Cargo.toml [hunk 1]
```diff
@@ -17,6 +17,7 @@ pallet-proxy = { git = "https://github.com/paritytech/substrate", branch = "polk
 primitives = { package = "acala-primitives", path = "../../primitives", default-features = false }
 sp-runtime = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.8", default-features = false }
 sp-std = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.8", default-features = false }
+chainbridge = { git = "https://github.com/AcalaNetwork/chainbridge-substrate", default-features = false, optional = true }
```
Comment (xlc) [inline modules/nft/Cargo.toml:20]: why nft need chainbridge?

File: runtime/benchmarks/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+[package]
+name = "benchmarks"
+version = "1.1.3"
```
Comment (xlc) [inline runtime/benchmarks/Cargo.toml:3]:
```suggestion
version = "1.2.0"
```

File: runtime/benchmarks/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+authors = ["Acala Developers"]
+edition = "2018"
+
+[dependencies]
+sp-io = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7", default-features = false }
```
Comment (xlc) [inline runtime/benchmarks/Cargo.toml:8]:
```suggestion
sp-io = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.8", default-features = false }
```

**Review by shaunxw — COMMENTED** [review]
Wrapping all benchmarks in macro would make them difficult to impl and debug.

As the only difference is the imports, we can use different features to control which runtime to import from, similar to how `sp-std` handles imports. In this way the benchmarks don't have to be in macro.

File: modules/nft/Cargo.toml (hunk 1 above)
Comment (ermalkaleci) [reply modules/nft/Cargo.toml:20]: It's only if you run benchmark tests inside directory that's why it's optional.

**Comment by ermalkaleci** [comment]
> Wrapping all benchmarks in macro would make them difficult to impl and debug.
> 
> 
> 
> As the only difference is the imports, we can use different features to control which runtime to import from, similar to how `sp-std` handles imports. In this way the benchmarks don't have to be in macro.

@shaunxw Not sure how to do that without ending up importing each other

**Comment by xlc** [comment]
so it will be:

karura-runtime -> common-benchmark (features = with-karura-runtime)
mandala-runtime -> common-benchmark (features = with-mandala-runtime)

File: modules/nft/Cargo.toml (hunk 1 above)
Comment (xlc) [reply modules/nft/Cargo.toml:20]: but that's wrong, it is something else need chainbridge, not nft

**Comment by ermalkaleci** [comment]
> so it will be:
> 
> karura-runtime -> common-benchmark (features = with-karura-runtime)
> mandala-runtime -> common-benchmark (features = with-mandala-runtime)

@xlc how do I import karura-runtime into common-benchmark?

File: modules/nft/Cargo.toml (hunk 1 above)
Comment (ermalkaleci) [reply modules/nft/Cargo.toml]: removed

**Comment by xlc** [comment]
You can use `include!("./common/benchmark.rs")` in karura-runtime to import the file

https://doc.rust-lang.org/std/macro.include.html

**Comment by ermalkaleci** [comment]
I don't see how this will make thing easier. It will be just a directory with rust code which have no purpose unless been included into runtime. Although we will get rid of the wrapper macro

**Comment by xlc** [comment]
You can have the code in mandala-runtime, and include it from karura-runtime.
And get rid of the wrapper macro is already a benefit. It helps IDE.

**Review by xlc — APPROVED** [review]

**Pull request merged**